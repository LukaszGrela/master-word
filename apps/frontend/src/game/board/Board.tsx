import React, { useMemo } from 'react';
import { classNames } from '@repo/utils';
import { IBoard } from '../types';
import './Board.scss';

/* Grid Board */
const Board: React.FC<IBoard> = ({
  children,
  columns,
  rows,
  className,
}): JSX.Element => {
  const style = useMemo(
    () => ({
      gridTemplateColumns: `repeat(${columns}, 1fr)`,
      gridTemplateRows: `repeat(${rows}, 1fr)`,
      width: `calc(calc(${
        columns - 1
      } * var(--gd-board-gap)) + calc(${columns} * var(--gd-board-size)))`,
    }),
    [columns, rows],
  );

  return (
    <div
      data-testid="board"
      className={classNames('board', className)}
      style={style}
    >
      {children}
    </div>
  );
};

export default Board;
