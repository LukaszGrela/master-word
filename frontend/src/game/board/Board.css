.board {
  display: grid;
  gap: 0.25rem;
  padding: 0.25rem;
  background-color: silver;
}
