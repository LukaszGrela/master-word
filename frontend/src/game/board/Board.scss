:root {
  --gd-board-size: 4rem;
  --gd-board-gap: 0.25rem;
  --gd-board-padding: 0.25rem;

  @media screen and (max-width: 320px) {
    --gd-board-size: 3rem;
  }
  @media screen and (min-width: 321px) and (max-width: 360px) {
    --gd-board-size: 3.5rem;
  }
}

.board {
  display: grid;
  gap: var(--gd-board-gap);
  padding: var(--gd-board-padding);
  background-color: transparent;
  @media (prefers-color-scheme: dark) {
    background-color: transparent;
  }
}
