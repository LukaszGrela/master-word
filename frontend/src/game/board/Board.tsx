import React from 'react';
import { IBoard } from '../types';
import { classNames } from '../../utils/classNames';
import './Board.scss';

/* Grid Board */
const Board: React.FC<IBoard> = ({
  children,
  columns,
  rows,
  className,
}): JSX.Element => {
  const style = {
    gridTemplateColumns: `repeat(${columns}, 1fr)`,
    gridTemplateRows: `repeat(${rows}, 1fr)`,
    width: `calc(calc(${
      columns - 1
    } * var(--gd-board-gap)) + calc(${columns} * var(--gd-board-size)))`,
  };

  return (
    <div className={classNames('board', className)} style={style}>
      {children}
    </div>
  );
};

export default Board;
