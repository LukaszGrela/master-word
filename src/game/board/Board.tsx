import React from 'react';
import { IBoard } from '../types';
import './Board.css';

/* Grid Board */
const Board: React.FC<IBoard> = ({ children, columns, rows }): JSX.Element => {
  const style = {
    gridTemplateColumns: `repeat(${columns}, 1fr)`,
    gridTemplateRows: `repeat(${rows}, 1fr)`,
    width: `calc(calc(${columns - 1} * 0.25rem) + calc(${columns} * 4rem))`,
  };

  return (
    <div className='board' style={style}>
      {children}
    </div>
  );
};

export default Board;
