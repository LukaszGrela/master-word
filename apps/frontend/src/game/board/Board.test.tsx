import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';

import Board from './Board';

describe('game', () => {
  describe('board', () => {
    describe('Board', () => {
      it('Renders with required props', () => {
        const { container } = render(<Board columns={2} rows={2} />);

        expect(container).toMatchSnapshot();
      });
      it('Renders with all props', () => {
        const { container } = render(
          <Board columns={2} rows={2} className="test">
            <p>Cell</p>
          </Board>,
        );

        expect(container).toMatchSnapshot();
      });
    });
  });
});
