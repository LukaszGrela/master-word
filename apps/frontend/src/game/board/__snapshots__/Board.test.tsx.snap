// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`game > board > Board > Renders with all props 1`] = `
<div>
  <div
    class="board test"
    data-testid="board"
    style="grid-template-columns: repeat(2, 1fr); grid-template-rows: repeat(2, 1fr);"
  >
    <p>
      Cell
    </p>
  </div>
</div>
`;

exports[`game > board > Board > Renders with required props 1`] = `
<div>
  <div
    class="board"
    data-testid="board"
    style="grid-template-columns: repeat(2, 1fr); grid-template-rows: repeat(2, 1fr);"
  />
</div>
`;
